web/src/main/java/com/epam/rft/atsy/web/controllers/NewApplicationPopupController.java

```diff
@@ -30,20 +41,43 @@ public ModelAndView loadPage() {
   }
 
   @RequestMapping(method = RequestMethod.POST, value = "/secure/new_application_popup")
-  public String saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result) {
-    if (!result.hasErrors()) {
-      stateDTO.setStateType("newstate");
-      stateDTO.setStateIndex(0);
-
-      ApplicationDTO applicationDTO = ApplicationDTO.builder()
-          .creationDate(new Date())
-          .candidateId(stateDTO.getCandidateId())
-          .positionId(stateDTO.getPosition().getId())
-          .channelId(stateDTO.getChannel().getId())
-          .build();
-
-      applicationsService.saveApplicaton(applicationDTO, stateDTO);
+  public String saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result)
+      throws ObjectNotFoundException {
+    if (result.hasErrors()) {
```

Comment:
Use bean validation and send a meaningful response, see example at https://github.com/epam-debrecen-rft-2015/atsy/blob/master/web/src/main/java/com/epam/rft/atsy/web/controllers/rest/SingleCandidateController.java